lib/move/index.js

```diff
@@ -35,7 +35,9 @@ function move (source, dest, options, callback) {
   }
 
   function doRename () {
-    if (overwrite) {
+    if (path.resolve(source) === path.resolve(dest)) {
+      setTimeout(callback, 1)
+    } else if (overwrite) {
```

Comment:
`fs.move` is async do you want it to exit early as such too (the `setTimeout`).
Do y'all have a pattern in place for this already?

(I'll add tests too, just wanted to get the ball rolling)